src/utils.ts

```diff
@@ -72,14 +72,43 @@ function handleTokens(token: string): Token {
 	return { token: [{ proofs: obj.proofs, mint: obj?.mints[0]?.url ?? '' }] };
 }
 
-export async function deriveKeysetId(keys: MintKeys) {
+export function deriveKeysetId(keys: MintKeys) {
 	const pubkeysConcat = Object.entries(keys)
 		.sort((a, b) => +a[0] - +b[0])
 		.map(([, pubKey]) => pubKey)
 		.join('');
 	const hash = sha256(new TextEncoder().encode(pubkeysConcat));
 	return Buffer.from(hash).toString('base64').slice(0, 12);
 }
+/**
+ * merge proofs from same mint,
+ * removes TokenEntrys with no proofs or no mint field
+ * and sorts proofs by id
+ *
+ * @export
+ * @param {Token} token
+ * @return {*}  {Token}
+ */
+export function cleanToken(token: Token): Token {
```

Comment:
would be great to have a test for that funtion!

Reply:
done

Reply:
thank you for reviewing my prs ❤️